tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/PulsarGenericObjectSinkTest.java

```diff
@@ -97,67 +101,61 @@ public void testGenericObjectSink() throws Exception {
                     .topic(spec.outputTopicName)
                     .create();
             for (int i = 0; i < numRecords; i++) {
                 MessageId messageId = producer.newMessage()
                         .value(spec.testValue)
                         .property("expectedType", spec.schema.getSchemaInfo().getType().toString())
+                        .property("recordNumber", i + "")
                         .send();
                 log.info("sent message {} {}  with ID {}", spec.testValue, spec.schema.getSchemaInfo().getType().toString(), messageId);
             }
-        }
 
-        // wait that all sinks processed all records without errors
-        try (PulsarAdmin admin = PulsarAdmin.builder().serviceHttpUrl(container.getHttpServiceUrl()).build()) {
-            for (SinkSpec spec : specs) {
-                try {
-                    log.info("waiting for sink {}", spec.sinkName);
-                    for (int i = 0; i < 120; i++) {
-                        SinkStatus status = admin.sinks().getSinkStatus("public", "default", spec.sinkName);
-                        log.info("sink {} status {}", spec.sinkName, status);
-                        assertEquals(status.getInstances().size(), 1);
-                        SinkStatus.SinkInstanceStatus instance = status.getInstances().get(0);
-                        if (instance.getStatus().numWrittenToSink >= numRecords) {
-                            break;
-                        }
-                        assertTrue(instance.getStatus().numRestarts > 1, "Sink was restarted, probably an error occurred");
-                        Thread.sleep(1000);
-                    }
 
+            // wait that all sinks processed all records without errors
+
+            try {
+                log.info("waiting for sink {}", spec.sinkName);
+                for (int i = 0; i < 120; i++) {
                     SinkStatus status = admin.sinks().getSinkStatus("public", "default", spec.sinkName);
                     log.info("sink {} status {}", spec.sinkName, status);
                     assertEquals(status.getInstances().size(), 1);
-                    assertTrue(status.getInstances().get(0).getStatus().numWrittenToSink >= numRecords);
-                    assertTrue(status.getInstances().get(0).getStatus().numSinkExceptions == 0);
-                    assertTrue(status.getInstances().get(0).getStatus().numSystemExceptions == 0);
-                    log.info("sink {} is okay", spec.sinkName);
-                } finally {
-                    dumpSinkLogs(spec);
+                    SinkStatus.SinkInstanceStatus instance = status.getInstances().get(0);
+                    if (instance.getStatus().numWrittenToSink >= numRecords) {
+                        break;
+                    }
+                    assertTrue(instance.getStatus().numRestarts > 1, "Sink was restarted, probably an error occurred");
+                    Thread.sleep(1000);
```

Comment:
why we need sleep 1 second here, it will make this test cost at least 10 mins.

Reply:
As soon as the record are processed we exit the loop, usually you make only one cycle, because it is pretty immediate to receive the records on the sink.

BTW I am thinking to rework this test and submit the sink only once, connecting it to multiple topics

WDYT?

Reply:
Thank, I see.
What about moving the check that below `// wait that all sinks processed all records without errors` out of the loop

Reply:
I have reworked the test